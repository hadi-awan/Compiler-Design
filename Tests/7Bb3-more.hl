// Test functions with different types of parameters and return values

function sumAll(A)
  var i,sum;
  for i in A do
    sum = sum+i;
    end;
  return sum;
  end;

sumAll([1,10]);

function MaxSingleton(A)
  var i,max;
  for i in A do
    if i>max then max = i; fi;
    end;
  return {max};
  end;

MaxSingleton([1,10]*[5,15]);

function PickParity(A,#even)
  var i,NewSet;
  for i in A do
    if #even then
      if i%2==0 then NewSet = NewSet + {i}; fi;
    else
      if i%2==1 then NewSet = NewSet + {i}; fi;
    fi;      
    end;
  return NewSet;
  end;

println "The set of the even elements of [1,15] is: ",PickParity([1,15],#1);
println "The set of the odd elements of [1,15] is: ",PickParity([1,15],#0);  
  

function printParity(A)
  function #isSumEven(A)
    return sumAll(A)%2 == 0;
    end;
  if #isSumEven(A) then
    println "The sum of all the elements of ", A, " is even.";
  else     
    println "The sum of all the elements of ", A, " is odd.";
  fi;
  end;

printParity([1,10]);
printParity({1,3,5,7,9,11});