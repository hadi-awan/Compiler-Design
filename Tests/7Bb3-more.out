55
{10}
The set of the even elements of [1,15] is: {2, 4, 6, 8, 10, 12, 14}
The set of the odd elements of [1,15] is: {1, 3, 5, 7, 9, 11, 13, 15}
The sum of all the elements of {1, 2, 3, 4, 5, 6, 7, 8, 9, 10} is odd.
The sum of all the elements of {1, 3, 5, 7, 9, 11} is even.
